examples/basic_path_extractor/src/main.rs

```diff
@@ -0,0 +1,109 @@
+//! A basic example application for working with the Gotham Path Extractor
+
+extern crate futures;
+extern crate gotham;
+extern crate hyper;
+extern crate mime;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+#[macro_use]
+extern crate gotham_derive;
+
+use hyper::{Response, StatusCode};
+
+use gotham::http::response::create_response;
+use gotham::router::Router;
+use gotham::router::builder::{build_simple_router, DefineSingleRoute, DrawRoutes};
+use gotham::state::State;
+
+
+/// `Product` struct
+///
+/// It contains only a `name` field for the sake of simplicity
+#[derive(Deserialize, StateData, StaticResponseExtender)]
+struct MyProduct {
+    name: String,
+}
+
+/// Function to handle the `GET` requests coming to `/widgets/:name`
+fn get_product_handler(mut state: State) -> (State, Response) {
+    // Extract `name` from `state`
+    let name = state.take::<MyProduct>().name;
```

Comment:
I feel like highlighting the [FromState](https://github.com/gotham-rs/gotham/blob/935ab297113918e050064eab1addcadc619e6ed9/gotham/src/state/from_state.rs) trait in our examples is a good idea, thus this would become:

```
let product = MyProduct::borrow_from(&state);
```